cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -405,15 +409,32 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 
 	// GenericLabels
 	node.Labels = cloudprovider.JoinStringMaps(node.Labels, buildGenericLabels(template, nodeName))
+
 	// NodeLabels
 	node.Labels = cloudprovider.JoinStringMaps(node.Labels, extractLabelsFromAsg(template.Tags))
 
+	node.Spec.Taints = extractTaintsFromAsg(template.Tags)
+
 	if nodegroupName, clusterName := node.Labels["nodegroup-name"], node.Labels["cluster-name"]; nodegroupName != "" && clusterName != "" {
 		klog.V(5).Infof("Nodegroup %s in cluster %s is an EKS managed nodegroup.", nodegroupName, clusterName)
-		// TODO: Call AWS EKS DescribeNodegroup API, check if keys already exist in Labels and do NOT overwrite
-	}
 
-	node.Spec.Taints = extractTaintsFromAsg(template.Tags)
+		// Call AWS EKS DescribeNodegroup API, check if keys already exist in Labels and do NOT overwrite
```

Comment:
Ah, you're right. I think I need to flip the arguments. But it might be safer just to do the loop here in case that JoinStringMaps ever changes. I'll update that.

Reply:
I switched to overwriting the Autoscaling group Tags with the EKS API labels, because it seems like EKS isn't sending the ASG Tags to Kubernetes itself so scale ups based on the ASG Tags are started by CAS but Kubernetes still doesn't see the nodes as matching the node selector so the pod is never successfully started.